### PR TITLE
Add support for selecting the Output Interface

### LINKED ISSUE #123: output interface
Hi, 

Could you please add the possibility to define output interface ?
I am using this to monitor multiple wan links.

Thanks.

### PR DESCRIPTION
**Pull request by AKARSHITJOSHI**
Closes #123

### CODE REVIEW

File: tcping.go [hunk 1]
```diff
@@ -632,9 +639,36 @@ func (tcpStats *stats) handleConnSuccess(rtt float32, connTime time.Time) {
...
+		if ipAddr = addr.(*net.IPNet).IP; ipAddr != nil {
+			break
+		}
+	}
+	if ipAddr == nil {
+		tcpStats.printer.printError("Unable to generate Interface's IP Address")
+	}
+	return ipAddr.String()
+}
+
```
Comment (pouriyajamshidi) [inline tcping.go:662]: I would also move this way up the code base, ideally below the `processUserInput` fucntion

File: tcping.go [hunk 2]
```diff
@@ -220,8 +222,8 @@ func processUserInput(tcpStats *stats) {
 	prettyJson := flag.Bool("pretty", false, "use indentation when using json output format. No effect without the '-j' flag.")
 	showVersion := flag.Bool("v", false, "show version.")
 	shouldCheckUpdates := flag.Bool("u", false, "check for updates.")
-
 	flag.CommandLine.Usage = usage
+	interfaceN := flag.String("i", "no-interface", "interface to dial the tcp connection")
```
Comment (pouriyajamshidi) [inline tcping.go:226]:
```suggestion
	interfaceName := flag.String("i", "no-interface", "interface to use for TCP connections")
```

**Review by pouriyajamshidi — CHANGES_REQUESTED** [review]

File: tcping.go [hunk 3]
```diff
@@ -632,9 +639,36 @@ func (tcpStats *stats) handleConnSuccess(rtt float32, connTime time.Time) {
...
+	return ipAddr.String()
+}
+
 // tcping pings a host, TCP style
 func tcping(tcpStats *stats) {
-	IPAndPort := netip.AddrPortFrom(tcpStats.userInput.ip, tcpStats.userInput.port)
+	var IPAndPort netip.AddrPort
+	if tcpStats.isInterface {
+		IPAdress, _ := netip.ParseAddr(getInterfaceAddr(tcpStats))
+		IPAndPort = netip.AddrPortFrom(IPAdress, tcpStats.userInput.port)
```
[the remark this answers is not in the document]
Comment (AKARSHITJOSHI) [reply tcping.go:668]: should I add a field netip.AddrPort to tcp.userInput ??

[the remark this answers is not in the document]
Comment (pouriyajamshidi) [reply tcping.go:668]: I would start with just the interface for now.

**Pull request closed without merging**